sycl/source/detail/platform_impl.cpp

```diff
@@ -209,6 +212,93 @@ static int filterDeviceFilter(std::vector<RT::PiDevice> &PiDevices,
   return StartingNum;
 }
 
```

Comment:
I am not sure if an explicit specialization is the right thing to do here. This function is only ever called with 2 sets of template arguments so perhaps we could have 2 overloads instead of a template. Thoughts from anyone?